zipline/finance/performance/period.py

```diff
@@ -193,15 +227,14 @@ def position_tracker(self, obj):
         # we only calculate perf once we inject PositionTracker
         self.calculate_performance()
 
-    def rollover(self):
-        self.starting_value = self.ending_value
-        self.starting_exposure = self.ending_exposure
-        self.starting_cash = self.ending_cash
-        self.period_cash_flow = 0.0
-        self.pnl = 0.0
-        self.processed_transactions = {}
-        self.orders_by_modified = {}
-        self.orders_by_id = OrderedDict()
+    def rollover(self, capital_change=0):
```

Comment:
I'm not exactly trying to treat a daily period in a cumulative as a 'subperiod`. To me, daily and cumulative periods are unrelated and treat capital changes differently. A 'subperiod` is just defined by a capital change that comes within a period, and this is applicable for both daily and cumulative. A subperiod in a cumulative period can be several days. `rollover` is not really a method meant to create a new subperiod, it's to make a new period. We `rollover` when we are at the end of a daily period and want to transfer those end values to the start values of the new period